006-securing-kafka-connect-rest-endpoint.md

```diff
@@ -0,0 +1,197 @@
+# Securing the Kafka Connect REST API endpoint
+
+## Current situation
+
+Currently, instances of Kafka Connect that are deployed by the Strimzi operators are configured with the default REST API endpoint settings. This means that the Kafka Connect REST API endpoint uses HTTP on port 8083 and that the Strimzi KafkaConnect, KafkaConnector and KafkaMirrorMaker2 operators make unsecured REST client calls based on this default configuration.
+
+The default network policies created by the Strimzi operators restrict incoming REST API calls to only allow access from the operator pod. If required, further network policies can be provided to override the default policy and allow wider access.
+
+## Motivation for change
+
+The Kafka Connect REST API endpoint is perhaps the last key endpoint that cannot currently be secured using Strimzi. Many users have requested the ability to secure the Kafka Connect REST API endpoint, for example:
+ - https://github.com/strimzi/strimzi-kafka-operator/issues/3229
+ - https://github.com/strimzi/strimzi-kafka-operator/issues/2161
+ - https://cloud-native.slack.com/archives/CMH3Q3SNP/p1577722052135000
+
+## Proposed changes
+
+This proposal adds a `rest` property to the KafkaConnect spec, allowing users to configure Connect REST API  endpoints that use HTTP or HTTPS. Client authentication is not addressed here, but there is scope for future proposals to address it.
+
+Simple example CR:
+
+```
+apiVersion: kafka.strimzi.io/v1beta1
+kind: KafkaConnect
+spec:
+  rest:
+  - tls: {}
+  ...
+```
+
+This CR would cause the operator to generate a certificate (signed by the cluster CA certificate) and a key, both stored in a secret for use by the operator clients and the Kafka Connect pods. The certificate and key would be mounted in each Kafka Connect pod, which would then create SSL truststores and keystores from the cert and key and set the following properties in the generated connect configuration to enable an HTTPS connection:
```

Comment:
The KafkaConnect has not hard coupling to any particular Kafka cluster. It can be used even with clusters which are not Strimzi based. So I do not think this can rely on cluster CA certificate. It would need to have its own CA.